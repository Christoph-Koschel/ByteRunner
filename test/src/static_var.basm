%entry main
%include "./test/src/natives.hasm"
%define msg "Hallo"
%define offset ""

; needs an address point on top of the stack
store_var:
    swap 1
    push 69
    write32
    ret

; needs an address point on top of the stack
load_var:
    swap 1
    read32
    swap 1
    ret

dump_mem:
    push 0
    push 20
    int dump_memory
    ret

main:
    call dump_mem
    push offset
    call store_var
    call dump_mem

    push 420
    push offset
    call load_var

    plusi
    int print_i64
    call dump_mem
    halt