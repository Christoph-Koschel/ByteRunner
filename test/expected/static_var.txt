48 61 6C 6C 6F 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
48 61 6C 6C 6F 45 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
489
48 61 6C 6C 6F 45 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
